Processor/ProductToFlatArrayProcessor.php

```diff
@@ -17,26 +22,44 @@
  */
 class ProductToFlatArrayProcessor extends BaseProductToFlatArrayProcessor
 {
+    /** @var array */
+    protected $attributesToExclude;
+
     /** @var ObjectDetacherInterface */
     protected $objectDetacher;
 
+    /** @var  array */
+    protected $associationTypesToExclude;
+
+    /** @var  AttributeGroupRepository */
+    protected $attributeGroupRepository;
+
+    /** @var  AssociationTypeRepository */
+    protected $associationTypeRepository;
+
     /**
      * @param Serializer                   $serializer
      * @param ChannelManager               $channelManager
      * @param array                        $mediaAttributeTypes
      * @param ProductBuilderInterface|null $productBuilder
      * @param ObjectDetacherInterface|null $objectDetacher
+     * @param AttributeGroupRepository     $attributeGroupRepository
+     * @param AssociationTypeRepository    $associationTypeRepository
      */
     public function __construct(
         Serializer $serializer,
         ChannelManager $channelManager,
         array $mediaAttributeTypes,
         ProductBuilderInterface $productBuilder = null,
-        ObjectDetacherInterface $objectDetacher = null
+        ObjectDetacherInterface $objectDetacher = null,
```

Comment:
Why is the objectDetacher optionnal ? Should I make my parameters optionnal too ?